File: binance4j-core/src/main/java/com/binance4j/core/event/ScheduledEvent.java
package com.binance4j.core.event;

import java.util.TimerTask;

/** An scheduled event calling a {@link TimerTask}. */
public interface ScheduledEvent {
	/** @return The event instantiation time in ms. */.
	long getInitTime();

	/** @return The event last call time in ms. */.
	long getLastCall();

	/** Cancels the event. */
	void cancel();

	/** @return The number of times the task was called. */.
	int getTicks();

	/** @return Tells if the last tick is the final one. */.
	boolean isFinalTick();

	/** @return Tells if the event is running. */.
	boolean isRunning();
}

File: src/main/kotlin/com/binance4j/connectors/core/event/ScheduledEvent.kt
/*
 * MIT License
 *
 * Copyright (c) 2022 Binance4j
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.binance4j.connectors.core.event

import java.util.*

/** A scheduled event calling a [TimerTask].  */
interface ScheduledEvent {
	/** Cancels the event.  */
	fun cancel()
	
	/** The event instantiation time in ms. */
	val initTime: Long
	
	/** The event last call time in ms. */
	val lastCall: Long
	
	/** The number of times the task was called. */
	val ticks: Int
	
	/** Tells if the last tick is the final one. */
	val isFinalTick: Boolean
	
	/** Tells if the event is running. */
	val isRunning: Boolean
}